scripts/sanitycheck.fsx

```
@@ -422,23 +422,25 @@ let SanityCheckNugetPackages () =
     //let solutions = Directory.GetCurrentDirectory() |> DirectoryInfo |> findSolutions
     //NOTE: we hardcode the solutions rather than the line above, because e.g. Linux OS can't build/restore iOS proj
     let solutions =
-        FsxHelper.RootDir.EnumerateFiles().Where (
+        FsxHelper.SourceDir.EnumerateFiles().Where (
             fun file ->
 
                 match Misc.GuessPlatform() with
 
                 // xbuild cannot build .NETStandard projects so we cannot build the non-Core parts:
                 | Misc.Platform.Linux when "msbuild" = Environment.GetEnvironmentVariable "BuildTool" ->
-                    file.Name = "gwallet.linux.sln"
+                    file.Name = "gwallet.linux-legacy.sln"
 
                 | Misc.Platform.Mac ->
-                    file.Name = "gwallet.mac.sln"
+                    file.Name = "gwallet.mac-legacy.sln"
 
                 | _ (* stockmono linux and windows *) ->
 
                     // TODO: have a windows solution file
-                    file.Name = "gwallet.core.sln"
+                    file.Name = "gwallet.core-legacy.sln"
         )
+    if not (solutions.Any()) then
+        raise <| FileNotFoundException()
```

@Mersho this is the wrong way to throw this exception; it needs to receive the file name that was not found as an argument, to make it 200% more understandable to the person that receives this exception in their face please (always be thinking about the future developer that you're punishing, please; this practice is called "maintainability").

---

there's no way to throw exception based on whether each file exists or not, I can only throw an exception when no solution was found. I'm gonna create a better exception msg. Something like "No solution was found to do sanity check."

---

You can throw that exception message with failwith, indeed. But I still want to see a FileNotFoundException thrown for when there is a solution file that doesn't exist. You can do that in each iteration.

---

Haha, that failwith never happens. Iterations are done on files found by EnumerateFiles(). They all exist.